lib/database/repositories/QcFlagRepository.js

```diff
@@ -156,92 +88,57 @@ class QcFlagRepository extends Repository {
-                GROUP_CONCAT( DISTINCT qcfv.flag_id ) AS verifiedFlagsList,
-                GROUP_CONCAT( DISTINCT qcf.id ) AS flagsList
-
-            FROM quality_control_flags AS qcf
-            INNER JOIN quality_control_flag_types AS qcft
-                ON qcft.id = qcf.flag_type_id
-            LEFT JOIN quality_control_flag_verifications AS qcfv
-                ON qcfv.flag_id = qcf.id
-            INNER JOIN quality_control_flag_effective_periods AS qcfep
-                ON qcf.id = qcfep.flag_id
-            INNER JOIN data_pass_quality_control_flag AS dpqcf
-                ON dpqcf.quality_control_flag_id = qcf.id
-            INNER JOIN (${GAQ_PERIODS_VIEW}) AS gaq_periods
-                ON gaq_periods.data_pass_id = dpqcf.data_pass_id
-            INNER JOIN global_aggregated_quality_detectors AS gaqd
-                ON gaqd.data_pass_id = gaq_periods.data_pass_id
-                    AND gaqd.run_number = gaq_periods.run_number
-                    AND gaqd.detector_id = qcf.detector_id
-                    AND gaq_periods.run_number = qcf.run_number
-                        AND (qcfep.\`from\` IS NULL OR UNIX_TIMESTAMP(qcfep.\`from\`) <= gaq_periods.\`from\`)
-                        AND (qcfep.\`to\`   IS NULL OR gaq_periods.\`to\` <= UNIX_TIMESTAMP(qcfep.\`to\`))
-
-            GROUP BY
-                gaq_periods.data_pass_id,
-                gaq_periods.run_number,
-                gaq_periods.\`from\`,
-                gaq_periods.\`to\`
-            `;
-
         const query = `
             SELECT
-                effectivePeriods.runNumber,
-                effectivePeriods.dataPassId,
-                effectivePeriods.bad,
-                SUM(effectivePeriods.mcReproducible) > 0 AS mcReproducible,
-                GROUP_CONCAT(effectivePeriods.verifiedFlagsList) AS verifiedFlagsList,
-                GROUP_CONCAT(effectivePeriods.flagsList) AS flagsList,
+                gaq_periods.runNumber,
+                gaq_periods.dataPassId,
+                gaq_periods.bad,
+                gaq_periods.badWhenMcReproducibleAsNotBad,
+                SUM(gaq_periods.mcReproducible) > 0 AS mcReproducible,
+                GROUP_CONCAT(gaq_periods.verifiedFlagsList) AS verifiedFlagsList,
+                GROUP_CONCAT(gaq_periods.flagsList) AS flagsList,
 
                 IF(
                     run.time_start IS NULL OR run.time_end IS NULL,
                     IF(
-                        effectivePeriods.\`from\` IS NULL AND effectivePeriods.\`to\`  IS NULL,
+                        gaq_periods.\`from\` IS NULL AND gaq_periods.\`to\`  IS NULL,
                         1,
                         null
                     ),
                     SUM(
-                        COALESCE(effectivePeriods.\`to\`, UNIX_TIMESTAMP(run.time_end))
-                      - COALESCE(effectivePeriods.\`from\`, UNIX_TIMESTAMP(run.time_start))
+                        COALESCE(gaq_periods.\`to\`, UNIX_TIMESTAMP(run.time_end))
```

Comment:
I am not sure, but I think here what you should use is `rct_time_start` and `rct_time_end`. I created a PR here: https://github.com/AliceO2Group/Bookkeeping/pull/1816